v2/api/devices/v1api20210702/structure.txt

```diff
@@ -0,0 +1,841 @@
+│   │   │   │   │   ├── Key: *string
+│   │   │   │   │   └── Value: *string
+│   │   │   │   ├── FallbackRoute: *Object (5 properties)
+│   │   │   │   │   ├── Condition: *string
+│   │   │   │   │   ├── EndpointNames: Validated<string[]> (2 rules)
+│   │   │   │   │   │   ├── Rule 0: MaxItems: 1
+│   │   │   │   │   │   └── Rule 1: MinItems: 1
+│   │   │   │   │   ├── IsEnabled: *bool
+│   │   │   │   │   ├── Name: *string
+│   │   │   │   │   └── Source: *Enum (1 value)
+│   │   │   │   │       └── "DeviceMessages"
+│   │   │   │   └── Routes: Object (5 properties)[]
+│   │   │   │       ├── Condition: *string
+│   │   │   │       ├── EndpointNames: Validated<string[]> (2 rules)
+│   │   │   │       │   ├── Rule 0: MaxItems: 1
+│   │   │   │       │   └── Rule 1: MinItems: 1
+│   │   │   │       ├── IsEnabled: *bool
+│   │   │   │       ├── Name: Validated<*string> (1 rule)
+│   │   │   │       │   └── Rule 0: Pattern: "^[A-Za-z0-9-._]{1,64}$"
+│   │   │   │       └── Source: *Enum (6 values)
+│   │   │   │           ├── "DeviceConnectionStateEvents"
+│   │   │   │           ├── "DeviceJobLifecycleEvents"
+│   │   │   │           ├── "DeviceLifecycleEvents"
+│   │   │   │           ├── "DeviceMessages"
+│   │   │   │           ├── "Invalid"
+│   │   │   │           └── "TwinChangeEvents"
+│   │   │   └── StorageEndpoints: map[string]Object (5 properties)
+│   │   │       ├── AuthenticationType: *Enum (2 values)
+│   │   │       │   ├── "identityBased"
+│   │   │       │   └── "keyBased"
+│   │   │       ├── ConnectionString: *string
+│   │   │       ├── ContainerName: *string
+│   │   │       ├── Identity: *Object (1 property)
+│   │   │       │   └── UserAssignedIdentity: *string
+│   │   │       └── SasTtlAsIso8601: *string
+│   │   ├── Sku: *Object (2 properties)
+│   │   │   ├── Capacity: *int
+│   │   │   └── Name: *Enum (7 values)
+│   │   │       ├── "B1"
+│   │   │       ├── "B2"
+│   │   │       ├── "B3"
+│   │   │       ├── "F1"
+│   │   │       ├── "S1"
+│   │   │       ├── "S2"
+│   │   │       └── "S3"
+│   │   └── Tags: map[string]string
+│   └── Status: Object (11 properties)
+│       ├── Conditions: conditions.Condition[]
+│       ├── Etag: *string
+│       ├── Id: *string
+│       ├── Identity: *Object (4 properties)
+│       │   ├── PrincipalId: *string
+│       │   ├── TenantId: *string
+│       │   ├── Type: *Enum (4 values)
+│       │   │   ├── "None"
+│       │   │   ├── "SystemAssigned"
+│       │   │   ├── "SystemAssigned, UserAssigned"
+│       │   │   └── "UserAssigned"
+│       │   └── UserAssignedIdentities: map[string]Object (2 properties)
+│       │       ├── ClientId: *string
+│       │       └── PrincipalId: *string
+│       ├── Location: *string
+│       ├── Name: *string
+│       ├── Properties: *Object (24 properties)
+│       │   ├── AllowedFqdnList: string[]
+│       │   ├── AuthorizationPolicies: Object (4 properties)[]
+│       │   │   ├── KeyName: *string
+│       │   │   ├── PrimaryKey: *string
```

Comment:
Should be secret and thus redacted? We might need to talk to them about their Swagger.